crates/sncast/src/main.rs

```diff
@@ -569,4 +585,31 @@ fn update_cast_config(config: &mut CastConfig, cli: &Cli) {
         ),
         clone_or_else!(cli.wait_timeout, config.wait_params.get_timeout()),
     );
+
+    config.url = clone_or_else!(cli.rpc_url, config.url);
```

Comment:
I think this is a sign that we should go the other way around. Instead of updating `CastConfig` and then creating a provider from it, we should create the provider in the command. If RPC is not present, try to get it from `snfoundry.toml`.